src/jsCore.ts

```diff
@@ -0,0 +1,509 @@
+      data: details.eventData,
+    },
+    contexts: details.contexts,
+  });
+}
+
+function trackConsentGrantedEvent(details: {
+  tracker: string | null;
+  eventData: ConsentGrantedProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    const built = buildConsentGranted({
+      id: details.eventData.documentId,
+      version: details.eventData.version,
+      description: details.eventData.documentDescription,
+      expiry: details.eventData.expiry,
+    });
+    tracker.track(built.event, details.contexts.concat(built.context));
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackConsentWithdrawnEvent(details: {
+  tracker: string | null;
+  eventData: ConsentWithdrawnProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    const built = buildConsentWithdrawn({
+      all: details.eventData.all,
+      id: details.eventData.documentId,
+      version: details.eventData.version,
+      description: details.eventData.documentDescription,
+    });
+    tracker.track(built.event, details.contexts.concat(built.context));
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackEcommerceTransactionEvent(details: {
+  tracker: string | null;
+  eventData: EcommerceTransactionProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  forTracker(details.tracker, (tracker) => {
+    tracker.track(
+      buildEcommerceTransaction({
+        orderId: details.eventData.orderId,
+        total: details.eventData.totalValue,
+        affiliation: details.eventData.affiliation,
+        tax: details.eventData.taxValue,
+        shipping: details.eventData.shipping,
+        city: details.eventData.city,
+        state: details.eventData.state,
+        country: details.eventData.country,
+        currency: details.eventData.currency,
+      }),
+      details.contexts
+    );
+  });
+  return <Promise<void>>Promise.resolve();
+}
+
+function trackDeepLinkReceivedEvent(details: {
+  tracker: string | null;
+  eventData: DeepLinkReceivedProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  return trackSelfDescribingEvent({
+    tracker: details.tracker,
+    eventData: {
+      schema:
+        'iglu:com.snowplowanalytics.mobile/deep_link_received/jsonschema/1-0-0',
+      data: details.eventData,
+    },
+    contexts: details.contexts,
+  });
+}
+
+function trackMessageNotificationEvent(details: {
+  tracker: string | null;
+  eventData: MessageNotificationProps;
+  contexts: EventContext[];
+}): Promise<void> {
+  return trackSelfDescribingEvent({
+    tracker: details.tracker,
+    eventData: {
+      schema:
+        'iglu:com.snowplowanalytics.mobile/message_notification/jsonschema/1-0-0',
+      data: details.eventData,
+    },
+    contexts: details.contexts,
+  });
+}
+
+function removeGlobalContexts(): Promise<void> {
+  return <Promise<void>>Promise.reject('Not implemented');
```

Comment:
for consistency with the other rejects (e.g. [here](https://github.com/snowplow/snowplow-react-native-tracker/blob/20f6000c6e37ae903abb75d7b421a373ef207f65/src/events.ts#L58)), we could also make the reason an instance of error

Reply:
Updated, thanks!